storagenode/orders/sender.go

```diff
@@ -137,15 +169,69 @@ func (sender *Sender) runOnce(ctx context.Context) (err error) {
 }
 
 // Settle uploads orders to the satellite.
-func (sender *Sender) Settle(ctx context.Context, satelliteID storj.NodeID, orders []*Info) {
+func (sender *Sender) Settle(ctx context.Context, satelliteID storj.NodeID, orders []*Info, ch chan archiveRequest) {
 	log := sender.log.Named(satelliteID.String())
-	err := sender.settle(ctx, log, satelliteID, orders)
+	err := sender.settle(ctx, log, satelliteID, orders, ch)
 	if err != nil {
 		log.Error("failed to settle orders", zap.Error(err))
 	}
 }
 
-func (sender *Sender) settle(ctx context.Context, log *zap.Logger, satelliteID storj.NodeID, orders []*Info) (err error) {
+func (sender *Sender) handleBatches(ctx context.Context, ch chan archiveRequest) {
+	err := sender.doHandleBatches(ctx, ch)
+	if err != nil {
+		sender.log.Error("failed to handle batches", zap.Error(err))
+	}
+}
+
+func (sender *Sender) doHandleBatches(ctx context.Context, ch chan archiveRequest) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	// In case anything goes wrong, discard everything from the channel.
+	defer func() {
+		for range ch {
+		}
+	}()
+
+	buffer := make([]archiveRequest, 0, cap(ch))
+
+	// Flush performs an archive on all of the requests in the buffer.
+	flush := func() (err error) {
+		txn, err := sender.orders.BeginArchive(ctx)
```

Comment:
This implementation would be clearer if you passed in `[]ArchiveRequest` to the database.

Reply:
I agree wholeheartedly.  I think the channels, etc., needlessly hard to follow.

Reply:
I don’t think the channels are going away. This is about exposing a transaction handle as compared to exposing an api that can perform multiple actions against the database.